bundles/org.openhab.binding.dbquery/src/main/java/org/openhab/binding/dbquery/internal/DatabaseBridgeHandler.java

```diff
@@ -0,0 +1,130 @@
+/**
+ * Copyright (c) 2010-2021 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.dbquery.internal;
+
+import java.util.Collection;
+import java.util.List;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.TimeUnit;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.openhab.binding.dbquery.action.DBQueryActions;
+import org.openhab.binding.dbquery.internal.domain.Database;
+import org.openhab.core.thing.Bridge;
+import org.openhab.core.thing.ChannelUID;
+import org.openhab.core.thing.ThingStatus;
+import org.openhab.core.thing.ThingStatusDetail;
+import org.openhab.core.thing.binding.BaseBridgeHandler;
+import org.openhab.core.thing.binding.ThingHandlerService;
+import org.openhab.core.types.Command;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Base implementation common to all implementation of database bridge
+ *
+ * @author Joan Pujol - Initial contribution
+ */
+@NonNullByDefault
+public abstract class DatabaseBridgeHandler extends BaseBridgeHandler {
+    private static final long RETRY_CONNECTION_ATTEMPT_TIME_SECONDS = 60;
+    private final Logger logger = LoggerFactory.getLogger(DatabaseBridgeHandler.class);
+    private Database database = Database.EMPTY;
+    private @Nullable ScheduledFuture<?> retryConnectionAttemptFuture;
+
+    public DatabaseBridgeHandler(Bridge bridge) {
+        super(bridge);
+    }
+
+    @Override
+    public void initialize() {
+        logger.trace("initialize bridge {}", getThing().getUID());
+        initConfig();
+
+        database = createDatabase();
+
+        connectDatabase();
+    }
+
+    private void connectDatabase() {
+        logger.debug("connectDatabase {}", database);
```

Comment:
IMHO this should be removed you don't need to log this, then log if it is successful or not successful as well.

Reply:
I do not completely agree with this for the following reasons:
- It doesn't do any warm to the user as to have it logged has to explicitly enable the debug level for the addon category.
- Connect to the database can come from initialization, retry attempts,...
- When a user reports a strange issue that is non easily reproducible, asking him to enable trace for the addon category is very useful as you can't debug his machine. I needed it in some difficult bugs in influxdb persistence addon and has been proven to be very useful.
- I see mainly all the persistence addons logging when connected, disconnected,...
- Finally, although is true that you can have the item status changelog, it's for all the items, isn't it? Having that info only for the addon, and with addon language, it's a valuable information

Reply:
Leave as is then if it will help fault find for users, as changing it to trace does nothing as most people run at INFO level.